bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1140 @@
+        }
+    }
+
+    /**
+     * Enum of potential fatal HTTP 2 session/stream errors.
+     */
+    private enum Http2Error {
+        CLOSED,
+        FAILURE,
+        TIMEOUT,
+        RESET,
+        IDLE,
+        GO_AWAY,
+        UNAUTHORIZED;
+    }
+
+    /**
+     * Private exception for handling HTTP 2 stream and session errors.
+     */
+    @SuppressWarnings("serial")
+    private static class Http2Exception extends ApiException {
+        public final Http2Error error;
+
+        public Http2Exception(Http2Error error) {
+            this(error, null);
+        }
+
+        public Http2Exception(Http2Error error, @Nullable Throwable cause) {
+            super("HTTP 2 stream " + error.toString().toLowerCase(), cause);
+            this.error = error;
+        }
+    }
+
+    /**
+     * Adapter for listening to HTTP 2 session status events.
+     *
+     * The session must be permanently connected, so it ignores onIdleTimeout() events.
+     * It also handles the following fatal events by notifying the containing class:
+     *
+     * <li>onClose()</li>
+     * <li>onFailure()</li>
+     * <li>onGoAway()</li>
+     * <li>onReset()</li>
+     */
+    private class SessionListenerAdapter extends Session.Listener.Adapter {
+
+        @Override
+        public void onClose(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.CLOSED));
+        }
+
+        @Override
+        public void onFailure(@Nullable Session session, @Nullable Throwable failure) {
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.FAILURE));
+        }
+
+        @Override
+        public void onGoAway(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.GO_AWAY));
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Session session) {
+            return false;
+        }
+
+        @Override
+        public void onPing(@Nullable Session session, @Nullable PingFrame frame) {
+            checkAliveOk();
+            if (Objects.nonNull(session) && Objects.nonNull(frame) && !frame.isReply()) {
+                session.ping(new PingFrame(true), Callback.NOOP);
+            }
+        }
+
+        @Override
+        public void onReset(@Nullable Session session, @Nullable ResetFrame frame) {
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.RESET));
+        }
+    }
+
+    /**
+     * Enum showing the online state of the session connection.
+     */
+    private static enum State {
+        /**
+         * Session closed
+         */
+        CLOSED,
+        /**
+         * Session open for HTTP calls only
+         */
+        PASSIVE,
+        /**
+         * Session open for HTTP calls and actively receiving SSE events
+         */
+        ACTIVE;
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(Clip2Bridge.class);
```

Comment:
You need it static because you use it in the static method `isClip2Supported`.
In that case, please name it `LOGGER`.
![image](https://github.com/openhab/openhab-addons/assets/10186704/fc1d6db6-d675-4bbb-9fdd-bbac4594fcde)

Reply:
Done